cmd/skaffold/app/cmd/config/util.go

```diff
@@ -33,22 +33,40 @@ import (
 const defaultConfigDir = ".skaffold"
 const defaultConfigFile = "config"
 
-func resolveKubectlContext() {
+// ResolveKubectlContext sets up the actual kube-context to be used. The order
+// of precedence is:
+// 1) command line
+// 2) skaffold.yml
+// 3) current context from kube-config
+func ResolveKubectlContext(schemaContext string) {
 	if kubecontext != "" {
 		return
 	}
 
+	if schemaContext != "" {
```

Comment:
1. this whole logic should not be under the `config` cmd. 
2. nit: `schemaContext` I would rename to `deployConfigContext`